test/fail_compilation/test_switch_error.d

```diff
@@ -24,6 +24,7 @@ void test1()
 TEST_OUTPUT:
 ---
 fail_compilation/test_switch_error.d(105): Error: undefined identifier `doesNotExist`
+fail_compilation/test_switch_error.d(105): Error: The condition expression in this switch statement failed to compile
```

Comment:
E.g. this error message is redundant to the already existing message w.r.t. the undefined identifier. I read it as "There's another error here" even though it's an addition to the existing error message. Might be improved by using `errorSupplemental` instead of `error`.

Reply:
It should be error supplemental the other way round but that would mean saving the error from the expression semantic which I can't be bothered to do tonight at least. An API like a gag but for basically grouping errors might be a nice addition.

In a sea of errors this isn't useless either. It looks redundant, but almost all errors are redundant in a certain sense, we aren't the target demographic for error messages.

Reply:
Indeed, we might not be the best candidates to evaluate error messages.

Offering all required informations is usually a good idea but also requires delicacy - too much noise is detrimental as well. This snippet is a good example because the error is completely unrelated to the `switch`
 statement,